### PR TITLE
[UpdatePage]add upload_pages_from_url sync and async

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/blob_client_async.py [hunk 1]
```diff
@@ -1492,6 +1492,121 @@ def __init__(
...
+                              range_start,
+                              range_end,
+                              source_range_start,
+                              source_content_md5=None,
+                              source_if_modified_since=None,
+                              source_if_unmodified_since=None, source_if_match=None, source_if_none_match=None,
+                              lease=None, if_sequence_number_lte=None, if_sequence_number_lt=None,
+                              if_sequence_number_eq=None, if_modified_since=None, if_unmodified_since=None,
+                              if_match=None, if_none_match=None, timeout=None
+                              ):
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/blob_client_async.py]: Same comment as for `append_block_from_url`, everything after `source_content_md5` can be combined in **kwargs

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/blob_client_async.py [hunk 2]
```diff
@@ -1492,6 +1492,121 @@ def __init__(
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace_async
+    async def upload_pages_from_url(self, source_url,  # type: str
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/blob_client_async.py]: Please add remaining type hints

File: sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py [hunk 3]
```diff
@@ -2268,6 +2268,177 @@ def upload_page( # type: ignore
...
+    def upload_pages_from_url(self, source_url,  # type: str
+                              range_start,
+                              range_end,
+                              source_range_start,
+                              source_content_md5=None,
+                              source_if_modified_since=None,
+                              source_if_unmodified_since=None, source_if_match=None, source_if_none_match=None,
+                              lease=None, if_sequence_number_lte=None, if_sequence_number_lt=None,
+                              if_sequence_number_eq=None, if_modified_since=None, if_unmodified_since=None,
+                              if_match=None, if_none_match=None, timeout=None
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py]: Same comment as for async operation

File: sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py [hunk 4]
```diff
@@ -2268,6 +2268,177 @@ def upload_page( # type: ignore
         except StorageErrorException as error:
             process_storage_error(error)
 
+    def _upload_pages_from_url_options(  # type: ignore
+            self, source_url,  # type: str
+            range_start,
+            range_end,
+            source_range_start,
+            source_content_md5=None,
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py]: missing type annotations

File: sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py [hunk 5]
```diff
@@ -2268,6 +2268,177 @@ def upload_page( # type: ignore
...
+            An ETag value, or the wildcard character (*). Specify this header
+            to perform the operation only if the resource's ETag does not match
+            the value specified. Specify the wildcard character (*) to perform
+            the operation only if the resource does not exist, and fail the
+            operation if it does exist.
+        :param ~azure.storage.blob.models.CustomerProvidedEncryptionKey cpk:
+            Encrypts the data on the service-side with the given key.
+            Use of customer-provided keys must be done over HTTPS.
+            As the encryption key itself is provided in the request,
+            a secure connection must be established to transfer the key.
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py]: not yet implemented here

File: sdk/storage/azure-storage-blob/tests/test_page_blob.py [hunk 6]
```diff
@@ -346,6 +361,469 @@ def test_update_page_unicode(self):
...
+        self.assertEqual(blob_properties.get('last_modified'), resp.get('last_modified'))
+
+    @record
+    def test_upload_pages_from_url_and_validate_content_md5(self):
+        # Arrange
+        source_blob_data = self.get_random_bytes(SOURCE_BLOB_SIZE)
+        source_blob_client = self._create_source_blob(source_blob_data, 0, SOURCE_BLOB_SIZE - 1)
+        src_md5 = StorageContentValidation.get_content_md5(source_blob_data)
+        sas = source_blob_client.generate_shared_access_signature(
+            permission=BlobPermissions.READ + BlobPermissions.DELETE,
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-blob/tests/test_page_blob.py]: >BlobPermissions.DELETE [](start = 46, length = 22)

why is delete permission needed?

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/blob_client_async.py (hunk 1 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/aio/blob_client_async.py]: Thanks for pointing this out!
While I'm still curious how you usually decide if you want to put a parameter in **kwargs or not

**Pull request merged**